staging/src/k8s.io/client-go/rest/client_test.go

```diff
@@ -283,6 +284,66 @@ func TestHttpMethods(t *testing.T) {
 	}
 }
 
+func TestHttpProxy(t *testing.T) {
+	ctx := context.Background()
+	testServer, fh, _ := testServerEnv(t, 200)
+	defer testServer.Close()
+
+	testProxyServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		to, err := url.Parse(req.RequestURI)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+		req.RequestURI = ""
+		req.URL = to
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
```

Comment:
proxy nit, copy all headers from resp to w?

Reply:
Done.